src/posthog-core.js

```diff
@@ -620,6 +612,15 @@ PostHogLib.prototype.capture = addOptOutCheckPostHogLib(function (event_name, pr
         properties: this._calculate_event_properties(event_name, properties, start_timestamp),
     }
 
+    if (event_name === '$pageview' && this.get_config('_capture_performance')) {
+        const performance = {
+            navigation: getPerformanceEntriesByType('navigation'),
+            paint: getPerformanceEntriesByType('paint'),
+            resource: getPerformanceEntriesByType('resource'),
+        }
+        data.properties.performance = { ...data.properties.performance, ...performance }
```

Comment:
Suggestion: Given that this is an internal property that won't be usable by outside users, prefix this with a $